### PR TITLE
[MM-27827] Cypress test for MM-T1190 - Terms of service appears after verifying email

### PR DESCRIPTION
**Pull request by fmunshi**
#### Summary
- Adds a cypress test to ensure the custom terms of service appears after verifying the email

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-27827
- Test link: https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T1190

### CODE REVIEW

File: e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        cy.apiUpdateConfig({
+            EmailSettings: {
+                RequireEmailVerification: false,
+            },
+            SupportSettings: {
+                CustomTermsOfServiceEnabled: false,
+            },
+        });
+    });
```
Comment (srkgupta) [inline e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js:65]: This is not required. We have a global before method which is executed before every test and it automatically resets these settings.

**Review by srkgupta — APPROVED** [review]
Thanks @fmunshi. Overall the test looks great and is working fine. Just one minor suggestion. Please fix the same before you merge the PR. Approving the PR

File: components/terms_of_service/terms_of_service.jsx [hunk 2]
```diff
@@ -156,7 +156,10 @@ export default class TermsOfService extends React.PureComponent {
                 <div>
                     <div className='signup-team__container terms-of-service__container'>
                         <div className={termsMarkdownClasses}>
-                            <div className='medium-center'>
+                            <div
+                                className='medium-center'
+                                data-testid='terms-of-service'
```
Comment (saturninoabril) [inline components/terms_of_service/terms_of_service.jsx:161]: I'd suggest for consistency
```suggestion
                                data-testid='termsOfService'
```

File: e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            testTeam = team;
+
+            // # Verify new user email
+            cy.apiVerifyUserEmailById(testUser.id);
+        });
+    });
+
+    it('MM-T1190 - Appears after creating new account and verifying email address', () => {
+        // # Verify new user email
+        cy.apiVerifyUserEmailById(testUser.id);
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js:39]: This is done in before hook too.

File: e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        cy.apiLogin(testUser);
+
+        // # Visit the test team town square
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // * Ensure that the terms of service text shows as expected
+        cy.findByTestId('terms-of-service').should('be.visible').and('contain.text', customTermsOfServiceText);
+
+        // * Ensure that the accept terms button is visible and click it
+        cy.get('#acceptTerms').should('be.visible').click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js:51]: It's missing verification of "You should be directed to the team..." after click.

File: e2e/cypress/support/api/terms_of_service.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        url: '/api/v4/terms_of_service',
+        method: 'POST',
+        body: {
+            text,
+        },
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response.body);
+    });
+});
```
Comment (saturninoabril) [inline e2e/cypress/support/api/terms_of_service.js:21]: 💯

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @fmunshi! I think what you've done in before hook as compared to the lengthy step 1 of the test case makes sense. I'll modify the test case to reflect "test data preparation". Looks good to me, except for few comments.

File: e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js (hunk 3 above)
Comment (fmunshi) [reply e2e/cypress/integration/system_console/compliance/custom_terms_of_service_spec.js:36]: removed it from the before hook, thanks

**Review by saturninoabril — APPROVED** [review]
Thanks @fmunshi! Tested and passed. LGTM.

**Pull request merged**